### ISSUE TITLE
V4 generator is deterministic on Neko

### ISSUE THREAD

**Issue by NQNStudios** [flashultra/uuid#3]
I've run into bugs because on Neko, Uuid.v4() always returns the same sequence of ids in each run-through of a program.

To reproduce this problem:

```
haxelib install ihx
haxelib install uuid
haxelib run ihx -neko -lib uuid
>> [uuid.Uuid.v1(),uuid.Uuid.v4()] 
Array<String> : [0f642390-1cd0-11b2-8cf0-19ff53c1a638,190b6fa6-6558-4f1a-b0e0-48466ef3f225]
>> [uuid.Uuid.v1(),uuid.Uuid.v4()] 
Array<String> : [0f666d80-1cd0-11b2-8cf0-452fa3a84843,10182a8d-d1af-4304-ae27-cf42283ebaa7]
>> exit
haxelib run ihx -neko -lib uuid
>> [uuid.Uuid.v1(),uuid.Uuid.v4()]
Array<String> : [0f733ec0-1cd0-11b2-8cf0-19ff53c1a638,190b6fa6-6558-4f1a-b0e0-48466ef3f225]
>> [uuid.Uuid.v1(),uuid.Uuid.v4()]
Array<String> : [0f78bd00-1cd0-11b2-8cf0-452fa3a84843,10182a8d-d1af-4304-ae27-cf42283ebaa7]
```

You can see that the v1() results are different in between program sessions, but the v4() results are the same. I expected v4() to return a unique id 99.999999999999999999999999% of the time, and never to see a collision in my lifetime, even between different program sessions.

**Comment by flashultra** [comment]
Thank you for reporting the bug.

The problem is with the following code 
```Std.int(haxe.Timer.stamp() * 1000)```
It's always return ```-2147483648``` ( or in hex ```FFFF FFFF 8000 0000```)  ( overflow ) which   maybe is correct. 

I will need to do some more test.

**Comment by NQNStudios** [comment]
I thought that this commit in scottPlusPlus's fork might fix the problem, but it didn't: https://github.com/scottPlusPlus/uuid/commit/36eb3c13dbc539206287aa1dc9cdeaf39ef40234

**Comment by flashultra** [comment]
I can't commit the fix right now, but it should be something like that 
  1) Replace ``` static var  rndSeed:Int ......```  with 
     static var rndSeed:Int64 = Int64.fromFloatt(Timer.stamp() * 1000); 
  2)  Replace ``` private static function splitmix64_seed(index:Int):Int64 ``` with
  private static function splitmix64_seed(index:Int64):Int64
 
This should fix the problem, but I can't make test and commit right now.

**Comment by NQNStudios** [comment]
For now, I'm working around the issue by using v1() instead `#if neko`.

**Comment by flashultra** [comment]
Fixed with version 2.3.2  
 For update : 
 ``` haxelib update uuid```
